src/api/article/controller.js

```diff
@@ -61,67 +64,173 @@ export const getOne = ({ params: { slug }, user }, res, next) =>
+                  Article.findOne({ _id: articleId })
+                    .populate('author', POPULATE_OPTIONS.author)
+                    .populate('brand', POPULATE_OPTIONS.brand)
+                    .populate('collectionId', POPULATE_OPTIONS.collection)
+                    .populate('locales', POPULATE_OPTIONS.locales)
+                    .then(serializeArticle)
+                    .then(sendJson(res))
+                    .catch(next)
+                )
+                .catch(next)
+            )
+            .catch(next)
+        )
+        .catch(next)
+    )
+    .catch(next);
 
-export const update = ({ params: { slug }, body }, res, next) =>
-  LocalizedArticle.findOne({ slug })
+export const update = ({ params: { slugOrId }, body }, res, next) =>
+  LocalizedArticle.findOne({ slug: slugOrId, active: true })
+    .then(result => (result && result.articleId) || (isValidId(slugOrId) && slugOrId))
     .then(checkIsFound)
-    .then(({ articleId }) =>
-      Article.findOneAndUpdate({ _id: articleId }, body, { new: true })
-        .populate('author', POPULATE_OPTIONS.author)
-        .populate('brand', POPULATE_OPTIONS.brand)
-        .populate('collectionId', POPULATE_OPTIONS.collection)
-        .populate('locales', POPULATE_OPTIONS.locales)
+    .then(articleId =>
+      ArticleBrand.findOne({ slug: body.brandSlug })
+        .then(newBrand =>
+          ArticleCollection.findOne({ slug: body.collectionSlug })
+            .then(newCollection =>
+              User.findOne({ email: body.authorEmail, role: 'author' })
+                .then(newAuthor => {
+                  const defaultFields = omit(body, [
+                    'author',
+                    'authorEmail',
+                    'brand',
+                    'brandSlug',
+                    'collectionSlug',
+                    'locales',
+                  ]);
+                  // TODO(uladbohdan): to refactor that, i.e. with pickBy / omitBy.
+                  if (newBrand) {
+                    defaultFields.brand = newBrand._id;
+                  }
+                  if (newCollection) {
+                    defaultFields.collectionId = newCollection._id;
+                  }
+                  if (newAuthor) {
+                    defaultFields.author = newAuthor._id;
+                  }
+
+                  return defaultFields;
+                })
+                .then(updateFields =>
+                  Article.findOneAndUpdate({ _id: articleId }, updateFields, { new: true })
+                    .then(checkIsFound)
+                    .then(async article => {
+                      // Here Article is properly updated. Proceeding with locales.
+                      let articleOldLocales = [];
+                      if (body.locales) {
+                        articleOldLocales = article.locales.slice(); // Copying.
+                        article.locales.splice(0, article.locales.length); // Clearing with no reassignment.
```

Comment:
??

Reply:
No reassignment is for linter to shut up.

Reply:
in such cases better use `ignore`